shared/cluster.go

```diff
@@ -409,3 +409,139 @@ func WriteDataPod(namespace string) (string, error) {
 
 	return RunCommandHost(cmd)
 }
+
+type ServiceType string
+
+const (
+	Server ServiceType = "server"
+	Agent  ServiceType = "agent"
+)
+
+func (s ServiceType) IsValidServiceType() bool {
+	valid := []ServiceType{Server, Agent}
+	for _, i := range valid {
+		if s == i {
+			return true
+		}
+	}
+
+	return false
+}
+
+type ServiceAction string
+
+const (
+	Stop    ServiceAction = "stop"
+	Start   ServiceAction = "start"
+	Restart ServiceAction = "restart"
+	Status  ServiceAction = "status"
+)
+
+func (s ServiceAction) IsValidServiceAction() bool {
+	valid := []ServiceAction{Stop, Start, Restart, Status}
+	for _, i := range valid {
+		if s == i {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (s ServiceAction) isStartAction() bool {
+	startAction := []ServiceAction{Start, Restart}
+	for _, i := range startAction {
+		if s == i {
+			return true
+		}
+	}
+
+	return false
+}
+
+type Product string
+
+const (
+	K3S  Product = "k3s"
+	RKE2 Product = "rke2"
+)
+
+func (p Product) IsValidProduct() bool {
+	valid := []Product{K3S, RKE2}
+	for _, i := range valid {
+		if p == i {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (p Product) getString() string {
```

Comment:
you are creating enums, to bellow have the need to create a method to return it to string again

same for the types

normally we use enums and his respective validations in specific situations to operate in sensitive data, or have a place like source of truth of your growing size of data model, like having 15 products  and actions that will potentially grow or modelling data to use as json payload, dbs , message systems and etc 

So my suggestion would be to not over-engineer on this that is very straightforward stuff , like just pass the desired action with your product to actually use in only one place "ManageClusterService"  
a simple switch or even map statement would be enough for all of those new methods ,enums and validations

Reply:
i hv removed the enums for the rest and used maps as MD and you suggested in another comment. After our conversation in the standup, moved the product stuff to product.go. this is expected to grow significantly with more actions - for example secret encryption, install, uninstall, and other ipv6 related stuff that MD was talking about in the standup as well.